Some small changes

mina86 opened the pull request:
I’m confused about the `contains` function. Is there something I’m missing? As far as I can tell, the grep check is sufficient.

mina86 committed:
remote-setup: add script remotely installing the lets-nfsn.sh

mina86 committed:
cron: use grep instead of implementing contains

mina86 committed:
cron: do not hard code path to letsencrypt.sh

mina86 committed:
setup: use printf and %q to write config file

mina86 committed:
setup: nfsn-hook.sh is executable already

Celti commented on nfsn-cron.sh:
```diff
@@ -20,8 +11,10 @@ git submodule update --remote
-if contains "Certificate will expire" "${checks[@]}"; then
+if find certs -name cert.pem -type l \
+	-exec openssl x509 -checkend 2592000 -in {} \; |
+	grep -F "Certificate will expire" >/dev/null
```
You should typically use `grep -q` instead of `grep >/dev/null`.

mina86 replied:
Done. I typically avoid `-q` since it’s not so portable. In this case we know NFSN has grep with `-q` support.

Celti replied:
`-q` is long-time standard POSIX. If that's not portable, what is? ^_^

Celti commented on nfsn-cron.sh:
```diff
@@ -17,8 +17,8 @@ if find certs -name cert.pem -type l \
-	echo "	cd ~/lets-nfsn.sh/letsencrypt.sh/"
-	echo "	./letsencrypt.sh --cron"
+	printf '	cd %q\n' "$PWD"
+	echo   '	./letsencrypt.sh --cron'
```
Actually, if we're doing that we can replace the whole thing with the following, and cut out the unnecessary `cd`:

``` sh
    printf '\t%q --cron\n' "$(realpath letsencrypt.sh)"
```

mina86 replied:
Done. I wasn’t sure if letsencrypt.sh will do the right thing when it’s not run from it’s own directory.

Celti replied:
It very happily does; on my servers it gets installed to `/usr/bin/`, reads global config in `/etc/`, and runs from wherever you please.

Celti commented on nfsn-setup.sh:
```diff
@@ -9,27 +9,24 @@ git submodule init
+# write_config $config_dir $wellknown
+write_config() {
+    mkdir -p -- "$1" "$2"
+    printf '%s=%q\n' WELLKNOWN "$2" HOOK "$(realpath nfsn-hook.sh)" >"$1/config"
```
Maybe should do something like `readonly hook_path="$(realpath nfsn-hook.sh)"` up at the top of the script.

mina86 replied:
Done.

Celti commented on nfsn-setup.sh:
```diff
@@ -9,27 +9,25 @@ git submodule init
-      echo " + Installing hook script..."
-      echo "HOOK='$(realpath nfsn-hook.sh)'" >> "${CONFIGDIR}/config"
-      chmod +x nfsn-hook.sh
+      write_config "letsencrypt.sh/certs/${site_root}/" \
```
One final nitpick: The trailing slash here results in a double-slash within the function as you write to `$1/config`, thus `${site_root}//config`.

mina86 replied:
Done.

Celti commented:
I don't think you're necessarily missing anything about `contains()` — I just find it more succinct and arguably clearer than the more-traditional pipe to grep. Either one is fine by me.

mina86 committed:
setup: cd to letsencrypt.sh

Most of the paths nfsn-setup.sh uses are relative to letsencrypt.sh
directory plus having it change directory to that repository brings
the script closer to nfsn-cron.sh

mina86 committed:
setup: letsencrypt.sh/.acme-challanges is no longer used

mina86 committed:
cron: type fix

mina86 committed:
hook: silence shellcheck warnings

The unused variables are present for documentation purposes so it’s
fine that they are not used.

tasken committed:
Update to use dehydrated

letsencrypt.sh had to change its name because of trademark issues.  Use
the new script.

mina86 committed:
Update to use dehydrated cont.

Replace last mentions of letsencrypt.sh which has been renamed to
dehydraded.

mina86 committed:
Deprecate the whole repository; point people to nfsn/lets-nfsn.sh
